Square Root Decomposition/MO's Algorithm/MOAlgo.cpp

```diff
@@ -0,0 +1,86 @@
+//Problem : http://www.spoj.com/problems/DQUERY/
+//Time Complexity : O((n+q)*sqrt(n))
+
+#include <bits/stdc++.h>
+
+#define ll long long int
+#define mod 1000000007
+#define show(a) for(i=0;i<a.size();i++) cout<<a[i]<<" ";
+#define fi first
+#define se second
```

Comment:
why not directly use `first` and `second`.

Reply:
Actually thats my template for Competitive Programming it saves time.Should i change it?

Reply:
@sirjan13 this repository is dedicated to all the learners and beginners, so please keep things straight and easily understandable.